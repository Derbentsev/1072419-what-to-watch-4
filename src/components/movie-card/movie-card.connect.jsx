import MovieCard from './movie-card';
import {connect} from 'react-redux';
import {ActionCreator} from '@components/app/reducer/reducer';


const mapDispatchToProps = (dispatch) => ({
  handleMovieCardClick(film) {
    dispatch(ActionCreator.setActiveFilm(film));
  },
});

export default connect(null, mapDispatchToProps)(MovieCard);
