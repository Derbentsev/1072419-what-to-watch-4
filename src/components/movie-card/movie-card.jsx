export const MovieCard = (props) => {
  const {film} = props;

  return (
    <article className="small-movie-card catalog__movies-card">
      <div className="small-movie-card__image">
        <img src={`img/${film.image}`} alt={film.title} width="280" height="175" />
      </div>
      <h3 className="small-movie-card__title">
        <a className="small-movie-card__link" href="movie-page.html">{film.title}</a>
      </h3>
    </article>
  );
};

MovieCard.propTypes = {
  film: PropTypes.shape({
    title: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
  }).isRequired,
};
