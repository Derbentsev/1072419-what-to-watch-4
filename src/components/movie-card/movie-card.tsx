import history from '@history/history';
import {film} from '@types/';


interface Props {
  handleMovieCardMouseEnter: () => {},
  renderPlayer: (string, string) => {},
  currentFilter: string,
  film: film,
}

const MovieCard: React.FunctionComponent<Props> = (props: Props) => {
  const {
    film,
    handleMovieCardMouseEnter,
    renderPlayer,
  } = props;

  return (
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={handleMovieCardMouseEnter}
      onClick={() => history.push(`/films/${film.id}`)}
    >

      {renderPlayer(film.previewVideoSrc, film.poster)}

      <h3 className="small-movie-card__title">
        <a className="small-movie-card__link" href="movie-page.html">
          {film.title}
        </a>
      </h3>
    </article>
  );
};


export default MovieCard;
