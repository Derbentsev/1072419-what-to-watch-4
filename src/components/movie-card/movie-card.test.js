import {MovieCard} from './movie-card';


const mock = {
  title: `title1`,
  image: `image1`,
};

it(`Render Movie Card`, () => {
  const onMovieCardMouseOverSpy = jest.fn();

  const tree = renderer
    .create(<MovieCard
      film = {mock}
      onMovieCardMouseOver = {onMovieCardMouseOverSpy}
    />)
      .toJSON();

  expect(tree).toMatchSnapshot();
});
