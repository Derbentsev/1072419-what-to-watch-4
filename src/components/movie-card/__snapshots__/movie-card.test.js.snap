// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Movie Card 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseOver={[MockFunction]}
>
  <div
    className="small-movie-card__poster"
  >
    <img
      alt="title1"
      height="175"
      onClick={[Function]}
      src="img/image1"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      title1
    </a>
  </h3>
</article>
`;
